### PR TITLE
feat(tooltip): add element customization

### PR DESCRIPTION
**Pull request by gloriliale**
- [ ] Added customization capability to Tooltip and Tooltip Text
- [ ] Create customization story and tests
- [ ] Included changeset

### CODE REVIEW

File: packages/paste-core/components/tooltip/src/index.tsx [hunk 1]
```diff
@@ -40,36 +37,46 @@ export interface TooltipStateReturn extends TooltipPrimitiveStateReturn {
...
-          </Text>
-        </StyledBase>
-      </TooltipPrimitive>
-    </>
-  );
-});
+/*
+Tooltip's current structure does not allow for customization of its arrow.
+TODO: refactor Tooltip so that the styling of its arrow can be customized
+using Customization Provider.
```
Comment (SiTaggart) [inline packages/paste-core/components/tooltip/src/index.tsx:48]: This is fine, can you just make a ticket it and put it under the customization elements epic please?

File: packages/paste-core/components/tooltip/stories/index.stories.tsx [hunk 2]
```diff
@@ -170,3 +171,32 @@ export const StateHookTooltip = (): React.ReactNode => {
...
+          <Button variant="primary">Open tooltip</Button>
+        </Tooltip>
+      </Box>
+    </CustomizationProvider>
+  );
+};
+
+CustomizedTooltip.story = {
+  name: 'Customized Tooltip',
+};
```
Comment (SiTaggart) [inline packages/paste-core/components/tooltip/stories/index.stories.tsx:202]: Don't we usually have a story to VRT a custom element name too?

File: packages/paste-core/components/tooltip/__test__/index.spec.tsx [hunk 3]
```diff
@@ -2,24 +2,30 @@ import * as React from 'react';
...
+import {CustomizationProvider} from '@twilio-paste/customization';
 // @ts-ignore typescript doesn't like js imports
 import axe from '../../../../../.jest/axe-helper';
 import {StateHookExample} from '../stories/index.stories';
 import {Tooltip} from '../src';
 
-const TooltipMock: React.FC = () => {
+expect.extend(matchers);
+
+const TooltipWithoutTheme: React.FC<{elementName?: string}> = ({elementName}) => {
```
Comment (TheSisb) [inline packages/paste-core/components/tooltip/__test__/index.spec.tsx:14]: 👍

**Review by TheSisb — APPROVED** [review]
After Simon's comments, lgtm

File: packages/paste-core/components/tooltip/stories/index.stories.tsx (hunk 2 above)
Comment (gloriliale) [reply packages/paste-core/components/tooltip/stories/index.stories.tsx:202]: I wasn't sure but just added one to tooltip and will keep that pattern in the future. Thanks!

File: packages/paste-core/components/tooltip/src/index.tsx (hunk 1 above)
Comment (gloriliale) [reply packages/paste-core/components/tooltip/src/index.tsx:47]: Donesee here

File: packages/paste-core/components/tooltip/src/index.tsx [hunk 4]
```diff
@@ -38,38 +35,47 @@ const StyledTooltip = React.forwardRef<HTMLDivElement, BoxProps>(({style, ...pro
 export interface TooltipStateReturn extends TooltipPrimitiveStateReturn {
   [key: string]: any;
 }
-export interface TooltipProps extends TooltipPrimitiveInitialState {
+export interface TooltipProps extends Pick<BoxProps, 'element'>, TooltipPrimitiveInitialState {
```
Comment (nkrantz) [inline packages/paste-core/components/tooltip/src/index.tsx:38]: Does the order matter here?

`export interface TooltipProps extends Pick<BoxProps, 'element'>, TooltipPrimitiveInitialState {`
vs.
`export interface TooltipProps extends TooltipPrimitiveInitialState, Pick<BoxProps, 'element'> {`

**Review by nkrantz — APPROVED** [review]
Lookin gooooood! Just left a couple questions for ya 🚀

File: packages/paste-core/components/tooltip/__test__/index.spec.tsx [hunk 5]
```diff
@@ -90,9 +96,84 @@ describe('Tooltip', () => {
...
+        <CustomizationProvider
+          // @ts-expect-error global test variable
+          theme={TestTheme}
+        >
+          <TooltipWithoutTheme />
+        </CustomizationProvider>
+      );
+      const tooltip = screen.getByTestId('tooltip-example');
+      const tooltipText = screen.getByText('Welcome to Paste!');
+      screen.debug();
```
[the remark this answers is not in the document]
Comment (gloriliale) [reply packages/paste-core/components/tooltip/__test__/index.spec.tsx:137]: yes it should!

File: packages/paste-core/components/tooltip/__test__/index.spec.tsx [hunk 6]
```diff
@@ -90,9 +96,84 @@ describe('Tooltip', () => {
...
+
+      expect(tooltip.getAttribute('data-paste-element')).toEqual('CUSTOM_TOOLTIP');
+      expect(tooltipText.getAttribute('data-paste-element')).toEqual('CUSTOM_TOOLTIP_TEXT');
+    });
+
+    it('should add custom styling to default Tooltip', (): void => {
+      render(
+        <CustomizationProvider
+          // @ts-expect-error global test variable
+          theme={TestTheme}
```
[the remark this answers is not in the document]
Comment (gloriliale) [reply packages/paste-core/components/tooltip/__test__/index.spec.tsx:130]: That's a good point, I think I was little confused on how to use the TestTheme. Is the following statement right?
> When we specify the theme as TestTheme, the aliases we use when specifying the styling here in the `elements=` part of CustomizationProvider refer to values of those aliases in the TestTheme (and those values will not change, which is why we made TestTheme in the first place).

File: packages/paste-core/components/tooltip/src/index.tsx (hunk 4 above)
Comment (gloriliale) [reply packages/paste-core/components/tooltip/src/index.tsx:38]: I'm not sure; the first time I did it I think I missed a comma or something so it gave me an error, and when it worked in this order, I kept it 😬 . Just changed it to have Pick<BoxProps, 'element'> be last for consistency though (which, it works now 😅 )

File: packages/paste-core/components/tooltip/__test__/index.spec.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (gloriliale) [reply packages/paste-core/components/tooltip/__test__/index.spec.tsx:130]: styles have been added!

**Pull request merged**